### ISSUE TITLE
in some case the shadow of imported scene won't be rendered if switch form GLES3 to GLES2

### ISSUE THREAD

**Issue by snougo** [godotengine/godot#44255]
**Godot version:**

3.2.4 beta3

**OS/device including version:**

![屏幕快照 2020-12-10 15 08 29](https://user-images.githubusercontent.com/64188282/101733309-9cfc9a00-3af9-11eb-83f5-32f141069afc.png)

**Issue description:**

if your whole scene is imported form outside, the shadow won't be rendered if you switch from GLES3 to GLES2, you have to restart Godot Editor then this problem will be fixed. also if your imported scene use some built-in mesh like cubemesh or spheremesh, then you won't be have this problem, shadow will be rendered normally when you switch from GLES3 to GLES2

**Steps to reproduce:**
find some outside model
import this model to the Godot
open the scene of this model and save is as .tscn
switch from GLES3 to GLES2
![屏幕快照 2020-12-10 15 23 29](https://user-images.githubusercontent.com/64188282/101734765-be5e8580-3afb-11eb-81a7-e6b77b5d1d4b.png)
![屏幕快照 2020-12-10 15 23 11](https://user-images.githubusercontent.com/64188282/101734774-c1f20c80-3afb-11eb-9b96-7d595c5d4615.png)

**Minimal reproduction project:**

[bug.report.zip](https://github.com/godotengine/godot/files/5670932/bug.report.zip)

**Comment by Calinou** [comment]
> the shadow won't be rendered if you switch from GLES3 to GLES2, you have to restart Godot Editor then this problem will be fixed.

I don't understand. You always have to restart the editor after switching renderers.

**Comment by snougo** [comment]
@Calinou 
yes, but the problem is after switching that shadow still not be rendered in editor and you have to restart again to fix this problem, closing scene and reopen it can fix this problem.

oh I forgot to say that in runtime the shadow works normally

**Comment by AlphaVDP2** [comment]
Reposting assets from closed issue for reference.

**Comment by AlphaVDP2** [comment]
Reposting assets from closed issue for reference.

Issue description:
No shadow is cast from a mesh when material is assigned to the Mesh instead of the Node.
Issue does not appear in GLES 3 mode.
Appears to be a regression as it worked in v3.2.3 in both GLES 2 and 3 modes.

Steps to reproduce:
Assign a material only to the Mesh and not the Node.

![shadow_bug](https://user-images.githubusercontent.com/79685520/109742510-835b8200-7c12-11eb-8b20-e9afda656469.png)

[Shadow_Bug.zip](https://github.com/godotengine/godot/files/6073194/Shadow_Bug.zip)

**Comment by snougo** [comment]
still happend in 3.4 beta5

**Comment by lawnjelly** [comment]
This broke somewhere between 3.2.4 beta 1 (it worked) and 3.2.4 beta 2 (didn't work).

**Comment by akien-mga** [comment]
Is this still reproducible in 3.5 beta 5 or later? I remember other issues caused by #43206 and some invalid RIDs that might have been solved by @lawnjelly and others.

**Comment by snougo** [comment]
@akien-mga 
yes it still reproducible in 3.5 beta5 on Mac, but I am not sure if it has be fixed in other Platforms like windows and linux

**Comment by akien-mga** [comment]
This is still reproducible in the current `3.x` branch (1b6ac2afb3509051aedd5981546255384e72a2f9) on Linux, with both Intel HD Graphics 630 and AMD Radeon RX Vega M. So it's likely not OS nor driver specific since so far it could be reproduced on all systems we tried with.

As https://github.com/godotengine/godot/issues/44255#issuecomment-789375705 describes it, assigning a Material to the MeshInstance directly or to the Mesh Surface seems to have an impact on whether the bug happens, though it might be more due to the order in which things are processed. For example in the "Shadow_Bug.zip" project, clearing the material on the "broken" mesh and then undoing it fixes the shadows.

**Comment by lawnjelly** [comment]
Curiously, the issue still occurs even after turning off "view grid" in the view menu. :thinking: Maybe we can try manually commenting out the grid code to confirm that it is causing the issue.
